include/types/tile_shape.hpp

```diff
@@ -5,13 +5,23 @@
 
 namespace tiledcuda::cell {
 
-template <size_t... T>
-using TileShape = cute::tuple<std::integral_constant<size_t, T>...>;
+template <size_t... Ns>
+struct TileShape {
+    static constexpr cute::array<size_t, sizeof...(Ns)> shape = {Ns...};
 
-template <const size_t I, typename TileShape>
-inline static constexpr size_t dim_size = cute::get<I>(TileShape{});
+    static constexpr size_t get_numel() {
+        size_t product = 1;
+        for (size_t n : shape) product *= n;
+        return product;
+    }
+
+    static constexpr size_t kNumel = get_numel();
+};
```

Comment:
This change is due to the original implementation caused some CuTe's internal errors after Cutlass was upgraded to v3.5.0, which I cannot easily fix.